barretenberg/cpp/src/barretenberg/ecc/fields/field.fuzzer.hpp

```diff
@@ -0,0 +1,1317 @@
+                    } else {
+                        // Zero elements should remain zero
+                        assert(field_internal_state[idx] == Field::zero());
+                        assert(uint_internal_state[idx] == numeric::uint256_t(0));
+                    }
+                }
+
+                if (with_debug) {
+                    info("BATCH_INVERT: start_index: ",
+                         start_index,
+                         " count: ",
+                         count,
+                         " non_zero_count: ",
+                         non_zero_elements.size());
+                }
+            }
+            return BATCH_INVERT_SIZE;
+        }
+    }
+
+    // Structure to hold parsed instruction data
+    struct ParsedInstruction {
+        Instruction instruction;
+        std::vector<uint8_t> data;
+        size_t size;
+    };
+
+    /**
+     * @brief Parse instructions from data buffer into a vector
+     * @param Data Pointer to the data buffer
+     * @param Size Size of the data buffer
+     * @param max_steps Maximum number of instructions to parse
+     * @return std::pair<std::vector<ParsedInstruction>, size_t> Vector of parsed instructions and bytes consumed
+     */
+    std::pair<std::vector<ParsedInstruction>, size_t> parse_instructions(const unsigned char* Data,
+                                                                         size_t Size,
+                                                                         size_t max_steps)
+    {
+        std::vector<ParsedInstruction> instructions;
+        size_t data_offset = 0;
+        size_t steps_parsed = 0;
+
+        // Skip settings if present
+        if (Size >= SETTINGS_SIZE) {
+            const VMSettings* settings_ptr = reinterpret_cast<const VMSettings*>(Data);
+            settings = *settings_ptr;
+            data_offset += SETTINGS_SIZE;
+        }
+
+        while (data_offset < Size && steps_parsed < max_steps) {
+            if (data_offset >= Size) {
+                break;
+            }
+
+            Instruction instruction = static_cast<Instruction>(Data[data_offset]);
+            size_t instruction_size = 0;
+
+            // Map the instruction to a valid instruction by taking modulo
+            constexpr size_t NUM_INSTRUCTIONS = static_cast<size_t>(Instruction::BATCH_INVERT) + 1; // Total number of
+                                                                                                    // instructions
+            uint8_t original_instruction = Data[data_offset];
+            instruction = static_cast<Instruction>(original_instruction % NUM_INSTRUCTIONS);
+
+            // Determine instruction size based on type
+            switch (instruction) {
+            case Instruction::SET_VALUE:
+                instruction_size = SET_VALUE_SIZE;
+                break;
+            case Instruction::ADD:
+                instruction_size = ADD_SIZE;
+                break;
```

Comment:
I find it weird that we have size decoding here and the switch statement above also returns size, feels it could be simpler